src/bdf_utils.jl

```diff
@@ -132,10 +132,15 @@ end
 function calc_Lagrange_interp(k,weights,t,ts,u_history,u)
   if t in ts
     i = searchsortedfirst(ts,t,rev=true)
-    return u_history[:,i]
+    for j in eachindex(u)
+      u[j] = u_history[j,i]
+    end
+    return u
   else
     for i in 1:k+1
-      @.. u += weights[i]/(t-ts[i])*u_history[:,i]
+      for j in eachindex(u)
+        u[j] += weights[i]/(t-ts[i])*u_history[j,i]
+      end
```

Comment:
I still see a lot of cases which are de-looped. Use `@view` to make the `@..` non-allocating. Changing to loops will make it not compatible with GPUs.

Reply:
Oh, I do not get this into considerations. I will add @view later. 

Every column of `u_history` saves one history point solution. If `u0` and `u` is an array, 
```julia 
@.. u += weights[i]/(t-ts[i])*u_history[:,i]
```
will not calculate the same results as the loop using linear index. That's why I change to loop here. I will think more about how to deal with this.